### PR TITLE
feat: mvi api wave 2 - upsert, similarity, and all metadata

### PR DESCRIPTION
**Pull request by malandis**
Updates the python client with the latest wave of MVI API changes. This includes:
- upsert instead of add: replace add_item_batch with upsert_item_batch
- option to retrieve all metadata on search
- choice of similarity metric on create index: euclidean similarity, inner product, and cosine similarity
    - defaults to cosine similarity

### CODE REVIEW

**Comment by malandis** [comment]
Adding for early review

File: src/momento/internal/aio/_vector_index_control_client.py [hunk 1]
```diff
@@ -37,12 +38,35 @@ def __init__(self, configuration: VectorIndexConfiguration, credential_provider:
...
-            request = ctrl_pb._CreateIndexRequest(index_name=index_name, num_dimensions=num_dimensions)
+
+            if similarity_metric == SimilarityMetric.EUCLIDEAN_SIMILARITY:
+                request = ctrl_pb._CreateIndexRequest(
+                    index_name=index_name,
+                    num_dimensions=num_dimensions,
+                    euclidean_similarity=ctrl_pb._CreateIndexRequest._EuclideanSimilarity(),
+                )
+            elif similarity_metric == SimilarityMetric.INNER_PRODUCT:
+                print("here")
```
Comment (nand4011) [inline src/momento/internal/aio/_vector_index_control_client.py:56]: Is this an errant debug statement?

Comment (malandis) [reply src/momento/internal/aio/_vector_index_control_client.py:56]: Good catch

Comment (malandis) [reply src/momento/internal/aio/_vector_index_control_client.py]: Fixed in https://github.com/momentohq/client-sdk-python/pull/383/commits/18d537f433286c406334ca77fa457407904e9a56

**Review by nand4011 — APPROVED** [review]
LGTM

**Review by nand4011 — APPROVED** [review]

**Pull request merged**